src/main/java/org/kohsuke/github/GHRepositoryRule.java

```diff
@@ -0,0 +1,667 @@
+                return new TypeReference<List<StatusCheckConfiguration>>() {
+                };
+            }
+        };
+        /**
+         * strict_required_status_checks_policy parameter
+         */
+        public static final BooleanParameter STRICT_REQUIRED_STATUS_CHECKS_POLICY = new BooleanParameter(
+                "strict_required_status_checks_policy");
+        /**
+         * name parameter
+         */
+        public static final StringParameter NAME = new StringParameter("name");
+        /**
+         * negate parameter
+         */
+        public static final BooleanParameter NEGATE = new BooleanParameter("negate");
+        /**
+         * operator parameter
+         */
+        public static final Parameter<Operator> OPERATOR = new Parameter<Operator>("operator") {
+            @Override
+            TypeReference<Operator> getType() {
+                return new TypeReference<Operator>() {
+                };
+            }
+        };
+        /**
+         * regex parameter
+         */
+        public static final StringParameter REGEX = new StringParameter("regex");
+        /**
+         * workflows parameter
+         */
+        public static final ListParameter<WorkflowFileReference> WORKFLOWS = new ListParameter<WorkflowFileReference>(
+                "workflows") {
+            @Override
+            TypeReference<List<WorkflowFileReference>> getType() {
+                return new TypeReference<List<WorkflowFileReference>>() {
+                };
+            }
+        };
+        /**
+         * code_scanning_tools parameter
+         */
+        public static final ListParameter<CodeScanningTool> CODE_SCANNING_TOOLS = new ListParameter<CodeScanningTool>(
+                "code_scanning_tools") {
+            @Override
+            TypeReference<List<CodeScanningTool>> getType() {
+                return new TypeReference<List<CodeScanningTool>>() {
+                };
+            }
+        };
+    }
+
+    /**
+     * Basic parameter for a ruleset.
+     *
+     * @param <T>
+     *            the type of the parameter
+     */
+    public abstract static class Parameter<T> implements Function<JsonNode, T> {
+
+        private final String key;
+
+        /**
+         * Get the parameter type reference for type mapping.
+         */
+        abstract TypeReference<T> getType();
+
+        /**
+         * Instantiates a new parameter.
+         *
+         * @param key
+         *            the key
+         */
+        protected Parameter(String key) {
+            this.key = key;
+        }
+
+        /**
+         * Gets the key.
+         *
+         * @return the key
+         */
+        String getKey() {
+            return this.key;
+        }
+
+        @Override
+        public T apply(JsonNode jsonNode) {
+            if (jsonNode == null) {
+                return null;
+            }
+            return GitHubClient.getObjectMapper().convertValue(jsonNode, getType());
```

Comment:
Is there a way to do this with the `ObjectReader` returned from [getMappingObjectReader](https://github.com/hub4j/github-api/blob/574a388e336efbed25c071077e9e5e45e741a93c/src/main/java/org/kohsuke/github/GitHubClient.java#L981) instead of adding a new method?